compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -2186,18 +2184,38 @@ impl<'tcx> TyCtxt<'tcx> {
         }
     }
 
-    /// Gets the attributes of a definition.
-    pub fn get_attrs(self, did: DefId) -> Attributes<'tcx> {
+    // FIXME(@lcnr): Remove this function.
+    pub fn get_attrs_unchecked(self, did: DefId) -> &'tcx [ast::Attribute] {
         if let Some(did) = did.as_local() {
             self.hir().attrs(self.hir().local_def_id_to_hir_id(did))
         } else {
             self.item_attrs(did)
         }
     }
```

Comment:
Hey @lcnr , if and when you get around to removing this, what will be the recommended way to query for multi-component attributes?

We have a compiler tool with a late lint pass that checks for the presence of some `foo::bar` attribute on items, and it seems like if you get rid of this function, I won't be able to query for that since `PartialEq<Symbol> for Path` wants the path to have exactly one component.

Reply:
I don't think so. If I `get_attrs` with `attr = Symbol("foo")`, then that calls `a.has_name("foo")` for each attribute, which does `a.path == "foo"`, and that `==` wants `path` to have exactly one component:

https://github.com/rust-lang/rust/blob/ebf95836e3cfc8c442cc71239324d947aafa3eac/compiler/rustc_ast/src/ast.rs#L103-L108

Reply:
ah... so we should maybe add something like `get_tool_attrs` or something? :thinking: we definitely want to keep supporting this in some way '^^

Reply:
Something like that, yeah.